python/tank/bootstrap/resolver.py

```diff
@@ -295,8 +296,60 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
         )
 
         for pc in pipeline_configs:
+            # We'll need to provide a descriptor object for the config if
+            # possible. Note that it's possible that we'll be returning a
+            # None for the config descriptor. It's up to other filtering
+            # operations to remove those if desired.
+            #
+            # As in resolve_shotgun_configuration, the order of precedence
+            # is as follows:
+            #
+            # 1. windows/linux/mac path
+            # 2. descriptor
+            # 3. sg_descriptor
             path = ShotgunPath.from_shotgun_dict(pc)
 
+            try:
+                current_os_path = path.current_os
+            except ValueError:
+                current_os_path = None
+
+            uri = pc.get("descriptor") or pc.get("sg_descriptor")
+
+            if path:
+                # Make sure that the config has a path for the current OS.
+                if current_os_path is None:
+                    log.debug("Config isn't setup for %s: %s", sys.path, pc)
+                    cfg_descriptor = None
+                else:
+                    cfg_descriptor = create_descriptor(
+                        sg_connection,
+                        Descriptor.CONFIG,
+                        dict(path=current_os_path, type="path"),
+                    )
+            elif uri:
+                log.debug("Using descriptor uri: %s", uri)
+                cfg_descriptor = create_descriptor(
+                    sg_connection,
+                    Descriptor.CONFIG,
+                    uri,
+                )
+            else:
+                # If we have neither a uri, nor a path, then we can't get
+                # a descriptor for this config.
+                log.debug("No uri or path found for config: %s", pc)
+                cfg_descriptor = None
+
+            # We add to the pc dict even if the descriptor is a None. It'll be
+            # the responsibility of the filter call below to remove any pcs
+            # that are deemed to be not useful.
+            if cfg_descriptor is None:
+                log.debug("Unable to create descriptor for config: %s", pc)
```

Comment:
Maybe you could add a simple 'continue' here instead of letting _filter_pipeline_configurations do the dirty work? I don't think there is any benefit of doing the cleanup later, right?